### PR TITLE
feat: decompiler export gltf with textures

### PR DESCRIPTION
**Pull request by pjoe**
Support exporting gltf with all mats/texs when using decompiler CLI.

This is done by "combining" `AdvancedGuiFileLoader` and `BasicVpkFileLoader`

### CODE REVIEW

File: Decompiler/Decompiler.cs [hunk 1]
```diff
@@ -848,7 +848,7 @@ private void DumpVPK(string parentPath, Package package, string type, Dictionary
                 return;
             }
 
-            var fileLoader = new BasicVpkFileLoader(package);
+            var fileLoader = new GameFileLoader(package);
```
Comment (xPaw) [inline Decompiler/Decompiler.cs:851]: I think this needs to pass in filename too?

File: ValveResourceFormat/IO/GameFileLoader.cs [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+                resource.Read(stream);
+                return resource;
+            }
+
+            return null;
+        }
+
+        public ShaderCollection LoadShader(string shaderName) => null;
+
+        public static Stream GetPackageEntryStream(Package package, PackageEntry entry)
```
Comment (xPaw) [inline ValveResourceFormat/IO/GameFileLoader.cs:436]: Let's not move this into lib code. This should be done in https://github.com/ValveResourceFormat/ValvePak/issues/10

Comment (pjoe) [reply ValveResourceFormat/IO/GameFileLoader.cs:436]: oh, where did that come from, my bad will fix

Comment (pjoe) [reply ValveResourceFormat/IO/GameFileLoader.cs:436]: hmmm seems like it is needed by IFileLoader?

**Pull request closed without merging**